fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/jpa/JPAConfig.java

```diff
@@ -70,29 +77,48 @@ public JPAConfig(RoutingDataSource dataSource, JpaProperties properties, ObjectP
     @DependsOn("tenantDatabaseUpgradeService")
     public LocalContainerEntityManagerFactoryBean entityManagerFactory(EntityManagerFactoryBuilder factoryBuilder) {
         Map<String, Object> vendorProperties = getVendorProperties();
-        customizeVendorProperties(vendorProperties);
-        return factoryBuilder.dataSource(getDataSource()).properties(vendorProperties).persistenceUnit("jpa-pu")
-                .packages("org.apache.fineract").jta(false).build();
+        String[] packagesToScan = getPackagesToScan();
+        return factoryBuilder.dataSource(getDataSource()).properties(vendorProperties).persistenceUnit("jpa-pu").packages(packagesToScan)
```

Comment:
Do we need the customizer? Wouldn't it be "easier" with a simple configuration property to pass the packages as an array of strings here?

Reply:
Not necessarily. If you look at the JpaBaseConfiguration class - provided by Spring - they do programmatic resolution of necessary packages. I can imagine a similar situation where you have some logic to figure out which packages are important.